design/design-doc-template-stacks-experience.md

```diff
@@ -0,0 +1,473 @@
+7. Edit `config/stack/manifests/resources/app.yaml` as appropriate.
+
+#### Adding a CRD
+
+The tooling will make it simpler to add a CRD from scratch. For example,
+the following would create a basic CRD in the appropriate folder, and
+would update the `dependsOn` field of the stack's `app.yaml`:
+
+```
+kubectl crossplane stack crd init WordpressInstance wordpressinstances wordpress.samples.stacks.crossplane.io
+```
+
+There will also be a convenience flag in the `stack init` command so
+that people starting from scratch can initialize the stack and the first
+CRD with a single command:
+
+```
+$ kubectl crossplane stack init --template mygroup/mystackname --init-crd
+> CRD name: WordpressInstance
+> CRD plural: wordpressinstances
+> CRD subdomain: wordpress.samples.stacks.crossplane.io
+```
+
+### Building and publishing
+
+To build and publish a stack, the standard steps for building and
+publishing a stack are used.
+
+Building:
+
+```
+kubectl crossplane stack build
+```
+
+Publishing:
+
+```
+kubectl crossplane stack publish
+```
+
+### Consuming
+
+Consuming has two steps: installation and creation. Installation is
+installing the stack into the Crossplane control cluster. Creation is
+creating a Kubernetes resource which the stack recognizes, so that the
+stack will do something in response to it. In the case of template
+stacks, that usually will look something like rendering a set of yamls
+and applying the rendered output to the cluster.
+
+#### Install
+
+Installation is much the same as installing any stack:
+
+```
+kubectl crossplane stack install myorg/mystack mystack
+```
+
+This installs the stack from the `myorg/mystack` image, using the name `mystack`.
+
+#### Create
+
+Instantiation is also very similar to any stack:
+
+```
+cat > my-instance.yaml <<EOF
+apiVersion: thing.samples.stacks.crossplane.io/v1alpha1
+kind: ThingInstance
+metadata:
+  name: thinginstance-sample
+spec:
+  myfield: myvalue
+EOF
+
+kubectl apply -f my-instance.yaml
+```
+
+The difference is that underneath, the ThingInstance will be used as the
+input for rendering a template.
+
+#### Delete
+
+When the ThingInstance is deleted, the corresponding resources will also
+be deleted. This is the same behavior as in any stack.
+
+```
+cat > my-instance.yaml <<EOF
+apiVersion: thing.samples.stacks.crossplane.io/v1alpha1
+kind: ThingInstance
+metadata:
+  name: thinginstance-sample
+spec:
+  myfield: myvalue
+EOF
+
+kubectl delete -f my-instance.yaml
+```
+
+The instance is also deleted if the stack is uninstalled.
+
+#### Uninstall
+
+Uninstall looks the same as any other stack:
+
+```
+kubectl crossplane stack uninstall mystack
+```
+
+#### Upgrade and update
+
+Upgrading a stack version, and updating an instance of an object managed
+by the stack, are out of scope of this document. However, one could
+imagine that the process of upgrading a stack version would change a
+version number on object instances which go with the stack, and that the
```

Comment:
Says they are out of scope and then sets up some interesting half-thoughts :horse_racing: 

> change a version number on object instances which go with the stack

Would this be spec, label, or annotation on the Foo resource? Or are you suggesting that the RenderedBar resources would carry the Stack version?

This may be helpful for Template Stacks to make version related changes to RenderedBar resources.